### PR TITLE
Feature 84/guilds

### PR DESCRIPTION
**Pull request by chrismckiernan**

### CODE REVIEW

File: server/src/main/java/com/objectcomputing/checkins/services/guild/GuildController.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+                .headers(headers -> headers.location(URI.create(String.format("%s/%s", request.getUri(), newGuild.getGuildid()))));
+    }
+
+    /**
+     * Load the current guilds into checkinsdb.
+     *
+     * @param guildsList, array of {@link GuildCreateDTO guild create dto} to load {@link Guild guild(s)}
+     */
+
+    @Post("/load")
```
Comment (hwilliams314) [inline server/src/main/java/com/objectcomputing/checkins/services/guild/GuildController.java:63]: are endpoints supposed to be verbs?

Comment (chrismckiernan) [reply server/src/main/java/com/objectcomputing/checkins/services/guild/GuildController.java]: Changed to nouns

File: server/src/test/java/com/objectcomputing/checkins/services/guild/GuildControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+
+    @Test
+    void testCreateAGuild() {
+        GuildCreateDTO guildCreateDTO = new GuildCreateDTO();
+        guildCreateDTO.setName("name");
+        guildCreateDTO.setDescription("description");
+
+        Guild g = new Guild(guildCreateDTO.getName(), guildCreateDTO.getDescription());
+
+        when(guildService.save(eq(g))).thenReturn(g);
```
Comment (hwilliams314) [inline server/src/test/java/com/objectcomputing/checkins/services/guild/GuildControllerTest.java:49]: where is this eq() coming from?

File: server/src/main/java/com/objectcomputing/checkins/services/guild/GuildCreateDTO.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import javax.validation.constraints.NotBlank;
+
+@Introspected
+public class GuildCreateDTO {
+    @NotBlank
+    @Schema(required = true, description = "name of the guild")
+    private String name;
+    @NotBlank
+    @Schema(required = true, description = "description of the guild")
+    private String description;
```
Comment (mkimberlin) [inline server/src/main/java/com/objectcomputing/checkins/services/guild/GuildCreateDTO.java:15]: ditto on the spacing here

File: server/src/main/java/com/objectcomputing/checkins/services/guild/GuildServicesImpl.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        guildsRepo.findAll().forEach(guilds::add);
+        if (name != null) {
+            guilds.retainAll(guildsRepo.findByNameIlike(name));
+        }
+        if (memberid != null) {
+            guilds.retainAll(guildMemberRepo.findByMemberid(memberid)
+                    .stream().map(GuildMember::getGuildid).map(gid -> guildsRepo.findById(gid).orElse(null))
+                    .filter(Objects::nonNull).collect(Collectors.toSet()));
+        }
+        return guilds;
```
Comment (mkimberlin) [inline server/src/main/java/com/objectcomputing/checkins/services/guild/GuildServicesImpl.java:64]: *THIS*

We need this approach recreated in a number of other services. Well done.

File: server/src/main/resources/db/V10__create_guilds_table.sql [hunk 5]
```diff
@@ -0,0 +1,15 @@
+drop table if exists guilds;
```
Comment (bhatiam12) [inline server/src/main/resources/db/V10__create_guilds_table.sql:1]: @chrismckiernan there is V10 in develop so please change it to V11 and other one to V12. Also if these are in its own PR then chances are they will be merged quickly and we will not have another case of missing versions.

File: server/src/main/java/com/objectcomputing/checkins/services/guild/Guild.java [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+    @Schema(description = "the id of the guild", required = true)
+    private UUID guildid;
+    @NotBlank
+    @Column(name = "name", unique = true)
+    @Schema(description = "name of the guild")
+    private String name;
+    @NotBlank
+    @Column(name = "description")
+    @Schema(description = "description of the guild")
+    private String description;
```
[the remark this answers is not in the document]
Comment (chrismckiernan) [reply server/src/main/java/com/objectcomputing/checkins/services/guild/Guild.java:34]: Done

File: server/src/main/java/com/objectcomputing/checkins/services/guild/GuildCreateDTO.java (hunk 3 above)
Comment (chrismckiernan) [reply server/src/main/java/com/objectcomputing/checkins/services/guild/GuildCreateDTO.java:16]: Done

File: server/src/main/resources/db/V10__create_guilds_table.sql (hunk 5 above)
Comment (chrismckiernan) [reply server/src/main/resources/db/V10__create_guilds_table.sql]: Done

**Pull request merged**